packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1659,16 +1679,40 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         }
     }
 
-    public async createDataStore(pkg: string | string[]): Promise<IFluidRouter> {
-        return this._createDataStore(pkg, false /* isRoot */);
+    public async createDataStore(pkg: string | string[]): Promise<IDataStore> {
+        const internalId = uuid();
+        return channelToDataStore(
+            await this._createDataStore(pkg, false /* isRoot */, internalId),
+            internalId,
+            this,
+            this.mc.logger);
     }
 
-    public async createRootDataStore(pkg: string | string[], rootDataStoreId: string): Promise<IFluidRouter> {
+    private async createRootDataStoreLegacy(pkg: string | string[], rootDataStoreId: string): Promise<IFluidRouter> {
         const fluidDataStore = await this._createDataStore(pkg, true /* isRoot */, rootDataStoreId);
         fluidDataStore.bindToContext();
         return fluidDataStore;
     }
 
+    public async createRootDataStore(pkg: string | string[], rootDataStoreId: string): Promise<IFluidRouter> {
+        return this._aliasingEnabled === true && this.attachState !== AttachState.Detached ?
+            this.createAndAliasDataStore(pkg, rootDataStoreId) :
+            this.createRootDataStoreLegacy(pkg, rootDataStoreId);
+    }
+
+    private async createAndAliasDataStore(pkg: string | string[], alias: string, props?: any): Promise<IFluidRouter> {
+        const internalId = uuid();
+        const dataStore = await this._createDataStore(pkg, false /* isRoot */, internalId, props);
+        dataStore.bindToContext();
+        const aliasedDataStore = channelToDataStore(dataStore, internalId, this, this.mc.logger);
+        const result = await aliasedDataStore.trySetAlias(alias);
+        if (!result) {
+            throw new GenericError("dataStoreAliasConflict");
```

Comment:
There needs to be a balance between not exposing PII (any user data, including alias name can be PII) and actionability of such errors, especially in local testing scenarios. Simple "dataStoreAliasConflict" error may not give enough context on where and how error happened.

I think error should contain ID (as a property) but marked as potentially containing PII.
We also talked with Mark about possibility (maybe, in some future) of having something like that:
message = "dataStoreAliasConflict on ID = ${id}"
id = { value: alias, tag: "UserData"}
toString: () => "dataStoreAliasConflict on ID = foo" (i.e. formats it such that good string prints out for local debugging)

Reply:
Yeah.. this is exactly why I didn't want to log anything here. The alias is coming right from the customer. The internal id can be logged. I wasn't aware of the userdata tag, it doesn't seem to have a lot of usage. I'll look into it 👍